Source/utils/soundsample.cpp

```diff
@@ -151,25 +188,76 @@ int SoundSample::SetChunk(ArraySharedPtr<std::uint8_t> fileData, std::size_t dwB
 		LogError(LogCategory::Audio, "Aulib::Stream::open (from SoundSample::SetChunk): {}", SDL_GetError());
 		return -1;
 	}
-
+#else
+	stream_ = std::make_unique<audsrv_adpcm_t>();
+	int success = audsrv_load_adpcm(stream_.get(), fileData.get(), dwBytes);
+	if (success != 0) {
+		LogError(LogCategory::Audio, "audsrv_load_adpcm (from SoundSample::SetChunk): {}", success);
+		return -1;
+	}
+	sampleId_ = stream_.get();
+#endif
 	return 0;
 }
 
 void SoundSample::SetVolume(int logVolume, int logMin, int logMax)
 {
+#ifndef PS2
 	stream_->setVolume(VolumeLogToLinear(logVolume, logMin, logMax));
+#else
+	volume_ = VolumeLogToLinear(logVolume, logMin, logMax) * MAX_VOLUME;
+	if (channel_ == -1)
+		return;
+
+	audsrv_adpcm_set_volume_and_pan(channel_, volume_, pan_);
+#endif
 }
 
 void SoundSample::SetStereoPosition(int logPan)
 {
+#ifndef PS2
 	stream_->setStereoPosition(PanLogToLinear(logPan));
+#else
+	pan_ = PanLogToLinear(logPan) * 100;
+	if (channel_ == -1)
+		return;
+
+	audsrv_adpcm_set_volume_and_pan(channel_, volume_, pan_);
+#endif
 }
 
 int SoundSample::GetLength() const
 {
+#ifndef PS2
 	if (!stream_)
 		return 0;
 	return std::chrono::duration_cast<std::chrono::milliseconds>(stream_->duration()).count();
+#else
+	size_t size = 0;
+	int pitch = 0;
+
+	if (stream_ != nullptr) {
+		size = stream_->size;
+		pitch = stream_->pitch;
+	} else if (!file_path_.empty()) {
+		AssetHandle handle = OpenAsset(file_path_.c_str(), size);
+		if (handle.ok()) {
+			size -= 16;
+			uint32_t buffer[3];
+			if (handle.read(buffer, sizeof(buffer))) {
+				pitch = buffer[2];
+			}
+		}

```

Comment:
It's a bit surprising to have `GetLength()` open a file.
This should probably be done when the `SoundSample` is constructed.